clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -521,6 +522,7 @@ public boolean coordinatorUnknown() {
     protected void coordinatorDead() {
         if (this.coordinator != null) {
             log.info("Marking the coordinator {} dead for group {}", this.coordinator, groupId);
+            client.failUnsentRequests(this.coordinator, GroupCoordinatorNotAvailableException.INSTANCE);
```

Comment:
I think this exception passes through to the OffsetCommitCallback in `ConsumerCoordinator.commitOffsetsAsync()`. I wonder if we should wrap it in a CommitFailedException in that case since it doesn't give much context to the user. This seems orthogonal to this patch, so we can probably do it separately.